### PR TITLE
[redfish_firmware]: Enabling IPV6 support

### LINKED ISSUE #510: Firmware Ansible Playbook
--> We are trying to update firmware via ansible play for iDRAC Server IPv6 . 
       firmware.exec file is from the local artifactory FYR.
       we are facing an error stating that. 
      "Resolution": "Execute InstallFromRepo method, and then retry the operation."
 
--> we are blocked with catalog XML file as to construct as well. While we are trying to construct the catalog using the module 
      dellemc.openmanage.ome_firmware_catalog -->  Not Supported IPv6

Tuesday 04 July 2023  13:25:20 +0000 (0:00:00.051)       0:00:00.051 **********
fatal: [localhost]: FAILED! => {"changed": false, "msg": "invalid literal for int() with base 10: 'fbc'"}
 
**--> Modules we are using :** 
- dellemc.openmanage.redfish_firmware -->  Not supported IPv6
- dellemc.openmanage.ome_firmware_catalog --> Not Supported IPv6
- dellemc.openmanage.idrac_firmware_info --> Supported IPv6
- dellemc.openmanage.idrac_firmware --> Supported IPv6

**Sample playbook :**
tasks:
    - name: Update firmware from repository on a HTTP
      dellemc.openmanage.idrac_firmware:
        idrac_ip: "<ipv6>"
        idrac_user: "root"
        idrac_password: "calvin"
        validate_certs: False
        catalog_file_name: "/data/tasks/Catalog.xml"
        share_name: “<IPv6>”
        reboot: True
        job_wait: True
        apply_update: True
 
 
**TASK [Update firmware from repository on a HTTP]** *************************************************************************************************
Tuesday 04 July 2023  13:11:43 +0000 (0:00:00.055)       0:00:00.055 **********
fatal: [localhost]: FAILED! => {"changed": false, "msg": {"error": {"@Message.ExtendedInfo": [{"Message": "The GetRepoBasedUpdateList method did not complete successfully.", "MessageArgs": [], [MessageArgs@odata.count](mailto:MessageArgs@odata.count): 0, "MessageId": "IDRAC.2.5.SUP028", "RelatedProperties": [], [RelatedProperties@odata.count](mailto:RelatedProperties@odata.count): 0, "Resolution": "Execute InstallFromRepo method, and then retry the operation.", "Severity": "Warning"}], "code": "Base.1.8.GeneralError", "message": "A general error has occurred. See ExtendedInfo for more information"}}}

Can someone please help us here

### PR DESCRIPTION
**Pull request by ABHISHEK-SINHA10**
IPV6 support for redfish_firmware

### CODE REVIEW

File: plugins/module_utils/redfish.py [hunk 1]
```diff
@@ -102,6 +105,22 @@ def __init__(self, module_params=None, req_session=False):
         self.root_uri = '/redfish/v1/'
         self._headers = {'Content-Type': 'application/json', 'Accept': 'application/json'}
 
+        try:
+            ip_addr, port = self.hostname, self.protocol
```
Comment (jagadeeshnv) [inline plugins/module_utils/redfish.py:109]: self.protocol??

File: plugins/modules/redfish_firmware.py [hunk 2]
```diff
@@ -57,8 +57,10 @@
     - "Husniya Hameed (@husniya_hameed)"
     - "Shivam Sharma (@Shivam-Sharma)"
     - "Kritika Bhateja (@Kritika_Bhateja)"
+    - "Abhishek Sinha (@ABHISHEK-SINHA10)"
 notes:
     - Run this module from a system that has direct access to Redfish APIs.
+    - This module supports both IPv4 and IPv6 address for *hostname*.
```
Comment (jagadeeshnv) [inline plugins/modules/redfish_firmware.py]: not hostname - has to be baseuri

File: plugins/module_utils/redfish.py (hunk 1 above)
Comment (ABHISHEK-SINHA10) [reply plugins/module_utils/redfish.py:109]: please refer line 104, also refer below page: 

https://docs.python.org/3.9/library/socket.html#socket.getaddrinfo 

"port is a string service name such as 'http', a numeric port number or None. By passing None as the value of host and port, you can pass NULL to the underlying C API."

**Comment by ABHISHEK-SINHA10** [comment]
checkmarx

image

**Review by jagadeeshnv — DISMISSED** [review]

**Comment by sachin-apa** [comment]
run e2e

**Review by sachin-apa — DISMISSED** [review]

**Comment by sachin-apa** [comment]
run e2e

**Review by Kritika-Bhateja-03 — DISMISSED** [review]

File: plugins/modules/redfish_firmware.py (hunk 2 above)
Comment (ABHISHEK-SINHA10) [reply plugins/modules/redfish_firmware.py]: updated

**Review by jagadeeshnv — DISMISSED** [review]

**Review by jagadeeshnv — APPROVED** [review]

**Comment by ABHISHEK-SINHA10** [comment]
Sonar report

image

**Review by jagadeeshnv — DISMISSED** [review]

**Comment by sachin-apa** [comment]
run e2e

**Review by sachin-apa — DISMISSED** [review]

**Comment by sachin-apa** [comment]
run e2e

**Comment by ansible-collections-svc** [comment]
Can one of the admins verify this patch?

**Comment by sachin-apa** [comment]
run e2e

**Comment by sachin-apa** [comment]
run e2e

**Review by sachin-apa — APPROVED** [review]

**Review by jagadeeshnv — APPROVED** [review]

**Review by rajshekarp87 — APPROVED** [review]
Validation completed

**Pull request merged**